### PR TITLE
Sphinx documentation site

### PR DESCRIPTION
**Pull request by michaelpj**
This is a POC of a documentation site using Sphinx, which is what the new Cardano docs initiative uses.

I think we should do this before we start seriously writing more docs, but it's WIP for now.

- I moved `docs` -> `notes`, since otherwise it was a bit confusing. 
    - Possibly we can put the doc site somewhere else or generally reorganize.
- I ported over the Plutus Tx tutorial.
    - Generally seems nice, although the literate programming is somewhat lacking. However, I found a nice way to get what we want using `literalinclude` to include delimited chunks of a separate source file.
    - Pandoc doesn't understand asciidoc, annoyingly, but going `asciidoc -[asciidoctor]-> docbook -[pandoc]-> ReST` worked fine.
- I ported over the glossary. 
    - Sphinx has a glossary directive, and linking to it seems to work fairly nicely.

### CODE REVIEW

File: doc/tutorials/PlutusTx.hs [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+-- Additional support for lifting.
+import Language.PlutusTx.Lift
+-- Builtin functions.
+import Language.PlutusTx.Builtins
+-- The Plutus Tx Prelude, discussed further below.
+import Language.PlutusTx.Prelude
+
+-- Setup for doctest examples.
+
+-- $setup
```
Comment (j-mueller) [inline doc/tutorials/PlutusTx.hs:21]: The setup code doesn't need to be in the generated documentation, maybe just put it in a separate BLOCK

**Comment by j-mueller** [comment]
> using literalinclude to include delimited chunks of a separate source file

Separating the code from the text also makes it easier to leave out some of the code that's required to make it compile but not relevant for the docs

**Comment by michaelpj** [comment]
I don't really understand the index yet. It looks like it only puts in entries for things in the glossary at the moment, it might require explicit markup to add more index terms.

Overall I think I maybe like this way of dealing with the code more than the literate style for the reasons that you mention.

**Comment by michaelpj** [comment]
I moved a few remaining useful bits from `plutus-tutorial` and murdered it.

I think this is ready to go now.

**Pull request merged**